spec/syntax.md

```diff
@@ -477,9 +478,10 @@ quoted-char    = %x0-5B         ; omit \
                / %x7D-D7FF      ; omit surrogates
                / %xE000-10FFFF
 
-unquoted       = unquoted-start *name-char
+unquoted       = (unquoted-start / negative-start) *name-char
 unquoted-start = name-start / DIGIT / "."
                / %xB7 / %x300-36F / %x203F-2040
+negative-start = "-" ( DIGIT / "." )
```

Comment:
Could this be combined?

```abnf
unquoted-start = name-start
               / DIGIT
               / "."
               / %B7 
               / %300-36F 
               / %203F-2040
               / ("-" ( DIGIT / "." ))
```

Also, remind me why we added the combining mark range of U+0300-036F to the quoted literal start? Wouldn't we be better off (given that unquoted literals are a **convenience**) in not adding them? Or U+203F and U+2040?

Reply:
> Could this be combined?

Yes, they could be combined. I separated them because you mention on Slack earlier that:
> This change has the unhappy aspect of making `unquoted-start` sometimes consume multiple characters. 

> Also, remind me why we added the combining mark range of U+0300-036F to the quoted literal start? Wouldn't we be better off (given that unquoted literals are a **convenience**) in not adding them? Or U+203F and U+2040?

It's because we've been trying to stay as close as possible to the [XML Nmtoken](https://www.w3.org/TR/xml/#NT-Nmtoken), which includes these characters. In other words, we currently have:

```abnf
unquoted-start = name-start / DIGIT / "."
               / %xB7 / %x300-36F / %x203F-2040
name-char  = name-start / DIGIT / "-" / "." / ":"
           / %xB7 / %x300-36F / %x203F-2040
```

and at least technically we could "simplify" these so that we had

```abnf
unquoted-start = name-start / DIGIT / "."
               / %xB7 / %x300-36F / %x203F-2040
name-char  = unquoted-start / "-" / ":"
```

But that seems like it would ultimately raise more questions.

Reply:
I see... it requires looking back at `Nmtoken` and its definition in XML. That makes it clear.

I did have a comment on Slack about not having the weird two-character side effect, but then today we discussed in teleconference that we were "okay" as a group with that. With that in mind, I thought maybe the single production would be clearer.

Reply:
I'm fine either way; the end result is the same.

Reply:
> It's because we've been trying to stay as close as possible to the [XML Nmtoken](https://www.w3.org/TR/xml/#NT-Nmtoken), which includes these characters.

I've asked before, and will repeat the question: what is the value of staying _close_ to XML Nmtoken but not matching it exactly? If there's going to be deviation (and unexplained deviation at that), what makes that a better foundation than e.g. [UAX 31 Identifiers](https://www.unicode.org/reports/tr31/#Default_Identifier_Syntax), or for that matter of dropping the pretense that this syntax has any relevant relationship to another one?

Reply:
@gibson042 The main reason we were sticking close to Nmtoken is for compatibility with names found in CLDR, although I tend to agree with you that "sorta consistent with" is not a really useful feature (and is probably a tripping hazard): we want to ensure that `name` values found in CLDR are valid without modification. The `unquoted` production is recycling the production (but I think not for any technical reason--see below)

@eemeli My "simplification" was meant to make top-level productions (what people are trying to understand) as logical as possible. I think it is cleaner to define all of the name-like constructions like:

```abnf
foo       = foo-start *foo-char
foo-start = ...
foo-char  = ...
```

Or, if recycling `name-char`:

```abnf
foo       = foo-start *name-char
foo-start = ...
bar       = bar-start *name-char
bar-start = ...
...
name-char = ...
```

I am going against my slack comment in suggesting we allow consuming two characters because it seems a simpler compromise (that we can call out with a comment) than having alternate `-start` productions.

To @gibson042's point, there is no real reason why we need to use `name-char` in `unquoted` except that we already know it doesn't have any whitespace or other troublesome characters in it and implementers probably already have some code to validate that set. Is that enough of a reason?

Reply:
We really do not need a N+1th incompatible specification for identifiers. It would be greatly preferable to either:

- use an identifier grammar production that exactly matches some existing standard
- use UAX 31 

Any other choice requires an extraordinarily strong justification, which has not been provided here.

Reply:
I've combined the rules as suggested in the first comment here, so closing this thread. Will reply separately about the identifier basis.